src/test/java/teammates/test/cases/webapi/DeleteFeedbackResponseActionTest.java

```diff
@@ -20,16 +41,232 @@ protected String getRequestMethod() {
+        };
+
+        DeleteFeedbackResponseAction a = getAction(params);
+        JsonResult r = getJsonResult(a);
+
+        assertEquals(HttpStatus.SC_OK, r.getStatusCode());
+
+        // response is deleted
+        assertNull(logic.getFeedbackResponse(typicalResponse.getId()));
+
+        ______TS("Typical success case, instructor");
+
+        loginAsInstructor(instructor1OfCourse1.getGoogleId());
+
+        params = new String[] {
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+                Const.ParamsNames.FEEDBACK_RESPONSE_ID, typicalResponse3.getId(),
+        };
+
+        a = getAction(params);
+        r = getJsonResult(a);
+
+        assertEquals(HttpStatus.SC_OK, r.getStatusCode());
+
+        // response is deleted
+        assertNull(logic.getFeedbackResponse(typicalResponse3.getId()));
     }
 
     @Test
     @Override
     protected void testAccessControl() throws Exception {
-        // TODO
+
+        ______TS("wrong giver type");
+
+        loginAsStudent(student1InCourse1.getGoogleId());
+        String[] wrongGiverTypeParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+                Const.ParamsNames.FEEDBACK_RESPONSE_ID, typicalResponse.getId(),
+        };
+
+        verifyCannotAccess(wrongGiverTypeParams);
+
+        ______TS("preview mode, cannot access");
+
+        loginAsInstructor(instructor1OfCourse1.getGoogleId());
+        String[] previewParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+                Const.ParamsNames.FEEDBACK_RESPONSE_ID, typicalResponse.getId(),
+                Const.ParamsNames.PREVIEWAS, instructor1OfCourse1.email,
+        };
+
+        verifyCannotAccess(previewParams);
+
+        ______TS("response in session not open, cannot access");
+
+        loginAsInstructor(instructor1OfCourse1.getGoogleId());
+        String[] sessionNotOpenParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+                Const.ParamsNames.FEEDBACK_RESPONSE_ID, responseInClosedSession.getId(),
+        };
+
+        verifyCannotAccess(sessionNotOpenParams);
+
+        ______TS("Response contains question not intended shown to instructor, "
+                + "moderated instructor should not be accessible");
+
+        loginAsInstructor(instructor1OfCourse1.googleId);
+        String[] invalidModeratedInstructorSubmissionParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+                Const.ParamsNames.FEEDBACK_RESPONSE_ID, typicalResponse2.getId(),
+                Const.ParamsNames.FEEDBACK_SESSION_MODERATED_PERSON, instructor1OfCourse1.getEmail(),
+        };
+        verifyCannotAccess(invalidModeratedInstructorSubmissionParams);
+
+        ______TS("Instructor moderates student's response, should not be accessible");
```

Comment:
Instructor **can** moderates student's response right? Why it can not be accessed? Give a clearer description?